### PR TITLE
Gcp new check/iam

### PR DESCRIPTION
**Pull request by arielkru**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.common.models.enums import CheckResult, CheckCategories
+
+NINETY_DAYS = {'d': 90, 'h': 2160, 'm': 129600, 's': 7776000}
+ONE_DAY = {'d': 1, 'h': 24, 'm': 1440, 's': 86400}
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py:5]: @arielkru According to terraform, [rotation_period](https://www.terraform.io/docs/providers/google/r/kms_crypto_key.html) will always be in seconds. Why the hassle?

File: checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        id = "CKV_GCP_43"
+        supported_resources = ['google_kms_crypto_key']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
+        if 'rotation_period' in conf.keys():
+            time_unit = conf['rotation_period'][0][-1]
+            time = int(conf['rotation_period'][0][:-1])
+            if ONE_DAY[time_unit] <= time <= NINETY_DAYS[time_unit]:
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py:19]: @arielkru Why does `time` have to be `>=` one day?

File: checkov/terraform/checks/resource/gcp/GoogleProjectAdminServiceAccount.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.common.models.enums import CheckResult, CheckCategories
+import re
+
+USER_MANAGED_SERVICE_ACCOUNT = re.compile ('.*@\d+\.iam\.gserviceaccount\.com$')
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/gcp/GoogleProjectAdminServiceAccount.py:5]: @arielkru This is a user managed service account:
`test2-218@gcp-bridgecrew-deployment.iam.gserviceaccount.com`
It doesn't fit this regex. Am I wrong?

Comment (arielkru) [reply checkov/terraform/checks/resource/gcp/GoogleProjectAdminServiceAccount.py:5]: according to gcp docs there's only one format for user managed service accounts:
https://cloud.google.com/iam/docs/service-accounts#user-managed

I'm not sure how it aligns with your service account. WDYT?

File: checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py (hunk 2 above)
Comment (arielkru) [reply checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py:19]: that's a must according to terraform (https://www.terraform.io/docs/providers/google/r/kms_crypto_key.html - `rotation_period`)
I added it to verify the time is not an empty value, which as far as I understand is an option.

File: checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py (hunk 1 above)
Comment (arielkru) [reply checkov/terraform/checks/resource/gcp/GoogleKMSRotationPeriod.py:5]: I've done it based on the the benchmark (which refers to gcloud), your'e right.

**Pull request merged**